build-system/tasks/extension-helpers.js

```diff
@@ -730,34 +732,57 @@ async function buildBentoExtensionJs(dir, name, options) {
  * configuration.
  * @param {string} dir
  * @param {string} name
+ * @param {string} mode
  * @param {Object} options
  * @return {Promise<string>}
  */
-async function getBentoFilename(dir, name, options) {
-  const filename = `${name}.js`;
+async function getBentoBuildFilename(dir, name, mode, options) {
+  const modes = {
+    'standalone': {
+      filename: `${name}.js`,
+      toExport: false,
+    },
+    'web-component': {
+      filename: 'web-component.js',
+      toExport: true,
+    },
+  };
+  const {filename, toExport} = modes[mode];
+  if (!filename) {
+    throw new Error(
+      `Unknown bento mode "${mode}" (${name}:${options.version})\n` +
+        `Expected one of: ${Object.keys(modes).join(', ')}`
+    );
+  }
+
   if (await fs.pathExists(`${dir}/${filename}`)) {
     return filename;
   }
-  const generatedSource = await generateBentoEntryPointSource(name, options);
-  const generatedFilename = `build/${name}.js`;
+  const generatedSource = await generateBentoEntryPointSource(
+    name,
+    toExport,
+    options
+  );
+  const generatedFilename = `build/${filename}`;
   await fs.outputFile(`${dir}/${generatedFilename}`, generatedSource);
   return generatedFilename;
 }
 
 /**
  * @param {string} name
+ * @param {string} toExport
  * @param {Object} options
  * @return {Promise<string>}
  */
-async function generateBentoEntryPointSource(name, options) {
+async function generateBentoEntryPointSource(name, toExport, options) {
   const css = options.hasCss
     ? await fs.readFile(`build/${name}-${options.version}.css`, 'utf8')
     : null;
 
   return dedent(`
     import {BaseElement} from '../base-element';
     
-    function defineElement() {
+    ${toExport ? 'export default' : ''} function defineElement() {
```

Comment:
nit: slightly nicer to keep the condition in one place.
```suggestion
    function defineElement() {
```